File: demo/src/main/java/com/example/demo/dto/AnnonceUpdateDTO.java
package com.example.demo.dto;

import com.example.demo.enums.Categorie;
import com.example.demo.enums.Disponibilite;
import lombok.Data;

@Data
public class AnnonceUpdateDTO {
    private String title;
    private String description;
    private double price;
    private Categorie category;
    private Disponibilite disponibilite;
    private Long livraisonId;
}

File: OccaPlace_Angular/src/app/dto/AnnonceUpdateDTO.ts
import {Disponibilite} from "../enum/Disponibilite";
import {Categorie} from "../enum/Categorie";

export interface AnnonceUpdateDTO {
  title?: string; 
  description?: string;
  price?: number;
  category?: Categorie;
  disponibilite?: Disponibilite;
  livraisonId?: number;
}
